#!/usr/bin/env python
PACKAGE = "stop_on_blue"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# Numeric values using sliders
# General syntax is:
#   gen.add('var_name', type, 0, 'var_name', default, low, high)

# Checkbox using boolean value
# General syntax is:
#   gen.add('var_namne', bool_t, 0, 'var_name', default)

# Mask Colors
gen.add('mask_l_hue',       double_t,   0, 'mask_l_hue',         0  , 0, 359) # corrected
gen.add('mask_h_hue',       double_t,   0, 'mask_h_hue',         359, 0, 359) # corrected
gen.add('mask_l_sat',       double_t,   0, 'mask_l_sat',         0  , 0, 255)
gen.add('mask_h_sat',       double_t,   0, 'mask_h_sat',         255, 0, 255)
gen.add('mask_l_lum',       double_t,   0, 'mask_l_lum',         0  , 0, 255)
gen.add('mask_h_lum',       double_t,   0, 'mask_h_lum',         255, 0, 255)

# Mask Dilation
gen.add('mask_dilate',      double_t,   0, 'mask_dialate',       0,   0, 20)

# Robot Speed
gen.add('speed',            double_t,   0, 'speed',              0,   0, 2)

# Median blur
gen.add('use_median_blur',      bool_t,     0, 'use_median_blur',       True)
gen.add('median_blur_amount',   int_t,      0, 'median_blur_amount',    3,   0,   10)

exit(gen.generate(PACKAGE, "stop_on_blue", "StopOnBlue"))
